src/compiler/checker.ts

```diff
@@ -14544,6 +14544,17 @@ namespace ts {
                 getIndexInfoWithReadonly(numberIndexInfo, readonly));
             spread.objectFlags |= ObjectFlags.ObjectLiteral | ObjectFlags.ContainsObjectOrArrayLiteral | ObjectFlags.ContainsSpread | objectFlags;
             return spread;
+
+            function errorTypeIfTooLarge(): Type | undefined {
+                if (left.flags & right.flags & TypeFlags.Union) {
+                    const resultSize = (left as UnionType).types.length * (right as UnionType).types.length;
+                    if (resultSize > 100000) {
+                        tracing.instant(tracing.Phase.Check, "getSpreadType_DepthLimit", { leftId: left.id, rightId: right.id });
+                        error(currentNode, Diagnostics.Spread_expression_produces_a_union_type_that_is_too_complex_to_represent);

```

Comment:
I think this error is going to be frustrating to encounter with it also providing a suggestion for a workaround. What is the general-form workaround? Casting something somewhere to `any`?

Reply:
I don't think there's a fix I'm sufficiently confident in to make it a quick fix.  I gave this a new error number so that it would be easier to document/google.

Reply:
I did wonder about looking inside the union to figure out whether there's a smaller span we could squiggle.

Reply:
For the real code where this arose, the mitigation I suggested was adding a cast to each spread with a pre-existing type that listed all possible properties (optionally).  I imagine `any` would also work, possible at the expense of completions and such.